src/kokkos/test/kokkos/TrackingRecHit2D_t.cc

```diff
@@ -0,0 +1,70 @@
+#include "KokkosCore/kokkosConfigCommon.h"
+#include "KokkosCore/kokkosConfig.h"
+
+#include "CondFormats/pixelCPEforGPU.h"
+#include "KokkosDataFormats/TrackingRecHit2DKokkos.h"
+
+namespace testTrackingRecHit2DKokkos {
+
+  template<typename MemorySpace>
+  void fill(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+
+    assert(hits.data());
+    auto &hits_ = *hits.data();
+
+    Kokkos::parallel_for("fill", Kokkos::RangePolicy<KokkosExecSpace>(0, 1024), KOKKOS_LAMBDA(const size_t i) {
+              assert(hits_.nHits() == 200);
+              if (i > 200) return;
+            });
+
+    return;
+  }
+
+  template<typename MemorySpace>
+  void verify(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+
+    assert(hits.data());
+
+    auto const &hits_ = *hits.data();
+
+    Kokkos::parallel_for("fill", Kokkos::RangePolicy<KokkosExecSpace>(0, 1024), KOKKOS_LAMBDA(const size_t i) {
+              assert(hits_.nHits() == 200);
+              if (i > 200) return;
+            });
+
+    return;
+  }
+
+  template<typename MemorySpace>
+  void runKernels(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+    assert(hits.data());
+
+    fill(hits);
+    verify(hits);    
+  }
+}  // namespace testTrackingRecHit2DKokkos
+
+namespace testTrackingRecHit2DKokkos {
+  template<typename MemorySpace>  
+  void runKernels(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits);
+
+}
+
+int main() {	
+
+  kokkos_common::InitializeScopeGuard kokkosGuard({KokkosBackend<KokkosExecSpace>::value});
+
+  {
+    auto nHits = 200;
+
+    Kokkos::View<pixelCPEforGPU::ParamsOnGPU, KokkosExecSpace> _cpeParams("cpeparams");
+    Kokkos::View<uint32_t*, KokkosExecSpace> _hitsModuleStart("hitsmodulestart");
```

Comment:
Both serial and CUDA versions of the test fail with
```
Constructor for Kokkos View 'hitsmodulestart' has mismatched number of arguments. Number of arguments = 0 but dynamic rank = 1 
Aborted
```
The constructor of `_hitsModuleStart` would either need one additional parameter for the size, or be changed to call the default constructor. The default constructor (also for `_cpeParams`) would be closer to the CUDA test which passes `nullptr` for both.

Reply:
@alexstrel Thanks for the update. Could you address this comment as well? I'm not able to test without.

Reply:
@makortel, sure. I was able to build without issues in my environment, thought.

Reply:
Right, it builds but neither serial or CUDA binaries of the test run.